src/components/MainHome.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import './MainHome.css';
+import LeftTabBar from './LeftTabBar';
+import CustomButton from './CustomButton';
+import Room from './Room';
+import openSocket from 'socket.io-client';
+import Constants from '../Constants';
+import * as Types from '../Types';
+import CustomTextInput from './CustomTextInput';
+import TrieSearch from 'trie-search';
+
+type MainHomeState = {
+    sendTo: string;
+    connectToRoom: boolean;
+    roomid: string;
+    socket: SocketIOClient.Socket;
+    displayName: Types.UserDisplay;
+    users: Types.UserDisplayMap;
+    searchResults: Array<Types.UserDisplay>;
+};
+
+let usersTrie: Record<string, any>;
+
+export default class MainHome extends Component<RouteComponentProps, MainHomeState> {
+    state: MainHomeState = {
+        sendTo: '',
+        connectToRoom: false,
+        roomid: '',
+        socket: openSocket(`${Constants.SERVER_HOST}:${Constants.SERVER_PORT}`),
+        displayName: { userid: '', displayName: '', color: '' },
+        users: {},
+        searchResults: [],
+    };
+
+    componentDidMount(): void {
+        this.updateSearchResults = this.updateSearchResults.bind(this);
+
+        // Get random username and a list of connected users
+        this.state.socket.emit(Constants.GET_DISPLAY_NAME);
+        this.state.socket.emit(Constants.GET_USERS);
+
+        // If path contains /home/roomid, then attempt to connect to roomid.
+        const path = window.location.pathname;
```

Comment:
Do you think it's possible to use router hooks to do this instead?

Reply:
Oh yes for sure!